src/libs/Violations/ViolationsUtils.ts

```diff
@@ -0,0 +1,195 @@
+                newTransactionViolations.push({name: 'missingCategory', type: 'violation', userMessage: ''});
+            }
+        }
+
+        if (policyRequiresTags) {
+            const hasTagOutOfPolicyViolation = transactionViolations.some((violation) => violation.name === 'tagOutOfPolicy');
+            const hasMissingTagViolation = transactionViolations.some((violation) => violation.name === 'missingTag');
+            const isTagInPolicy = Boolean(policyTags[transaction.tag]?.enabled);
+
+            // Add 'tagOutOfPolicy' violation if tag is not in policy
+            if (!hasTagOutOfPolicyViolation && transaction.tag && !isTagInPolicy) {
+                newTransactionViolations.push({name: 'tagOutOfPolicy', type: 'violation', userMessage: ''});
+            }
+
+            // Remove 'tagOutOfPolicy' violation if tag is in policy
+            if (hasTagOutOfPolicyViolation && transaction.tag && isTagInPolicy) {
+                newTransactionViolations = reject(newTransactionViolations, {name: 'tagOutOfPolicy'});
+            }
+
+            // Remove 'missingTag' violation if tag is valid according to policy
+            if (hasMissingTagViolation && isTagInPolicy) {
+                newTransactionViolations = reject(newTransactionViolations, {name: 'missingTag'});
+            }
+
+            // Add 'missingTag violation' if tag is required and not set
+            if (!hasMissingTagViolation && !transaction.tag && policyRequiresTags) {
+                newTransactionViolations.push({name: 'missingTag', type: 'violation', userMessage: ''});
+            }
+        }
+
+        return {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transaction.transactionID}`,
+            value: newTransactionViolations,
+        };
+    },
+    /**
+     * Gets the translated message for each violation type. Necessary because `translate` throws a type error if you
+     * attempt to pass it template strings, or members of a string literal union that contains both function and string
+     * members.
+     *
+     * The issue is with the type of {@link PhraseParameters} which is defined as
+     *   `type PhraseParameters<T> = T extends (...args: infer A) => string ? A : never[];`
+     *
+     *  When a union type is passed that _may_ include a key that returns  a `string` or a key that contains a
+     * `function`, the type returns `never`.
+     *
+     *  If you have only switch cases with params, and use a default case to catch the strings, it will throw an error.
+     *
+     * @param violation
+     * @param translate
+     */
+    getViolationTranslation(violation: TransactionViolation, translate: <TKey extends TranslationPaths>(phraseKey: TKey, ...phraseParameters: PhraseParameters<Phrase<TKey>>) => string) {
+        switch (violation.name) {
+            case 'missingCategory':
```

Comment:
The type checker ensures these string literals cannot be something random like 'carlos' right?

Reply:
Yeah -- they are type safe. Think of them as a more efficient enum that you don't need to import or call.